src/browser/api/system.js

```
@@ -679,23 +679,21 @@ exports.System = {
     },
 
     getSelectedPreloadScripts: function(preloadOption) {
-        const response = {};
-
         const missingRequiredScripts = preloadOption.reduce((urls, preload) => {
             if (!preload.optional && !(preload.url in preloadScriptsCache)) {
                 urls.push(preload.url);
             }
             return urls;
         }, []);
 
-        if (!missingRequiredScripts.length) {
-            const missingOptionalScript = '';
-            response.scripts = preloadOption.map(preload => preloadScriptsCache[preload.url] || missingOptionalScript);
-        } else {
-            response.error = `Execution of preload scripts canceled because of missing required script(s) ${JSON.stringify(missingRequiredScripts)}`;
+        if (missingRequiredScripts.length) {
+            const err = `Execution of preload scripts canceled because of missing required script(s) ${JSON.stringify(missingRequiredScripts)}`;
```

lets return an error Object here.